modulefiles/ufs_hercules.intel.lua

```diff
@@ -0,0 +1,54 @@
+help([[
+loads UFS Model prerequisites for Hercules/Intel
+]])
+
+prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-dev-20230825/envs/ufs-pio-2.5.10/install/modulefiles/Core")
```

Comment:
Same question here as on the C5 PR. This is pointing to a 'dev' spack-stack. Shouldn't it use a version'd stack, like 1.4.1? 

Also, what determines the compiler version? Shouldn't we be trying to match major versions of intel on all platforms (ie, 2021.x.x, 2022.x.x)?

Reply:
My understanding is that the version of the compiler used in the stack in determined by what is available on the machine. 
@ulmononian

Reply:
Does hercules show these as available?

intel-oneapi-compilers/2022.0.2
 intel-oneapi-compilers/2022.2.1 (D)
intel-oneapi-compilers/2023.1.0

Reply:
this version has been fully tested for both ufs and jedi. 

btw: the `stack-intel/2021.9.0` is only spack wrapper on `intel-oneapi-compilers/2023.1.0`. please see https://github.com/JCSDA/spack-stack/blob/27f79a38c3f8dd00cd8e1f135df66104fd4713c0/configs/sites/hercules/compilers.yaml#L1-L13, which shows the actual system compiler used to build this stack (specifically, lines 5-8).

Reply:
why is the stack-intel/2021.9.0 if the compiler is 2023.1.0?

Reply:
i am not exactly sure why, but it is just a nomenclature adopted in spack-stack between the `spec` and the system module the spack `spec` is based on. in this case, it just corresponds to the actual specific compiler version number, since `intel-oneapi-compilers/2023.1.0` uses `ifort/2021.9.0`.

Reply:
@BrianCurtis-NOAA The compiler is not 2023.1.0. This is part of Intel's very successful strategy of confusing customers. Under Intel oneAPI 2023.1.0, each component has a potentially different version number. The compilers are 2021.9.0. The MPI library in this rare case is I believe the same, but generally that's not true (it wasn't true for the last 2 years of oneAPI releases). tbb, mkl, all of these have different version numbers. The distinction is important because spack needs those numbers to compose the path to the component from the top-level oneAPI installation directory.